### ISSUE TITLE
consul.json template renders incorrectly

### ISSUE THREAD

**Issue by siddharthist** [CiscoCloud/microservices-infrastructure#888]
Here are two `consul.json` configuration files generated from the Consul role. If you look carefully at [line 7](https://github.com/CiscoCloud/microservices-infrastructure/blob/aeb30e6703b053d889c3326a369adb5429cac506/roles/consul/templates/consul.json.j2#L7) and [line 12](https://github.com/CiscoCloud/microservices-infrastructure/blob/aeb30e6703b053d889c3326a369adb5429cac506/roles/consul/templates/consul.json.j2#L12) of the template, it becomes clear that `"retry_join"` should never be set when `"bootstrap_expect"` is equal to 1. 

Does Consul update this configuration file while running? Why else might this be happening? It seems that the Jinja numerical comparison is failing. 
```
[root@worker-001 vagrant]# cat /etc/consul/consul.json
{
  "datacenter": "vagrantdc",
  "advertise_addr": "192.168.242.56",
  "node_name": "worker-001",
  "rejoin_after_leave": true,
  "domain": "consul",
  "retry_join": [ "192.168.242.55" ],
  "server": true,
  "bootstrap_expect": 1,
  "encrypt": "rRkd0ToIm5slWBsuew5Zew==",
  "ca_file": "/etc/consul/ssl/ca.cert",
  "cert_file": "/etc/consul/ssl/consul.cert",
  "key_file": "/etc/consul/ssl/consul.key",
  "verify_incoming": true,
  "verify_outgoing": true,
  "data_dir": "/var/lib/consul",
  "ui_dir": "/usr/share/consul-ui",
  "enable_syslog": true,
  "disable_remote_exec": true,
  "http_api_response_headers": {
    "Access-Control-Allow-Origin": "*"
  },
  "dns_config": {
    "allow_stale": true,
    "max_stale": "5s"
  }
}
[vagrant@control-01 ~]$ sudo cat /etc/consul/consul.json
{
  "datacenter": "vagrantdc",
  "advertise_addr": "192.168.242.55",
  "node_name": "control-01",
  "rejoin_after_leave": true,
  "domain": "consul",
  "retry_join": [ "192.168.242.55" ],
  "server": true,
  "bootstrap_expect": 1,
  "encrypt": "rRkd0ToIm5slWBsuew5Zew==",
  "ca_file": "/etc/consul/ssl/ca.cert",
  "cert_file": "/etc/consul/ssl/consul.cert",
  "key_file": "/etc/consul/ssl/consul.key",
  "verify_incoming": true,
  "verify_outgoing": true,
  "data_dir": "/var/lib/consul",
  "ui_dir": "/usr/share/consul-ui",
  "enable_syslog": true,
  "disable_remote_exec": true,
  "http_api_response_headers": {
    "Access-Control-Allow-Origin": "*"
  },
  "dns_config": {
    "allow_stale": true,
    "max_stale": "5s"
  }
}
```

**Comment by ChrisAubuchon** [comment]
It looks like ansible thinks that `consul_bootstrap_expect` is a string type.

Matching `defaults/main.yml`:
```yaml
- name: debug
  debug:
    msg: "{{ consul_bootstrap_expect > 1 }}"
```

```
TASK: [consul | debug] ******************************************************** 
ok: [control-01] => {
    "msg": "True"
}
ok: [worker-001] => {
    "msg": "True"
}
```

String comparison:
```yaml
- name: debug
  debug:
    msg: "{{ consul_bootstrap_expect > '1' }}"
```

```
TASK: [consul | debug] ******************************************************** 
ok: [control-01] => {
    "msg": "False"
}
ok: [worker-001] => {
    "msg": "False"
}
```

Typecasting to `int`:
```yaml
- name: debug
  debug:
    msg: "{{ consul_bootstrap_expect|int > 1 }}"
```

```
TASK: [consul | debug] ******************************************************** 
ok: [control-01] => {
    "msg": "False"
}
ok: [worker-001] => {
    "msg": "False"
}
```

I have no idea why it's different in the `vagrant-up` branch but I think typecasting to `int` is the better way to go

**Comment by siddharthist** [comment]
Great, thank you @ChrisAubuchon!!

**Comment by siddharthist** [comment]
Fixed in #942
